moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp

```diff
@@ -102,25 +102,27 @@ robot_model::RobotModelConstPtr getSharedRobotModel(const std::string& robot_des
   }
 }
 
-planning_scene_monitor::CurrentStateMonitorPtr getSharedStateMonitor(const robot_model::RobotModelConstPtr& kmodel,
+planning_scene_monitor::CurrentStateMonitorPtr getSharedStateMonitor(const std::string& robot_description,
+                                                                     const robot_model::RobotModelConstPtr& kmodel,
                                                                      const boost::shared_ptr<tf::Transformer>& tf)
 {
-  return getSharedStateMonitor(kmodel, tf, ros::NodeHandle());
+  return getSharedStateMonitor(robot_description, kmodel, tf, ros::NodeHandle());
 }
 
-planning_scene_monitor::CurrentStateMonitorPtr getSharedStateMonitor(const robot_model::RobotModelConstPtr& kmodel,
+planning_scene_monitor::CurrentStateMonitorPtr getSharedStateMonitor(const std::string& robot_description,
+                                                                     const robot_model::RobotModelConstPtr& kmodel,
                                                                      const boost::shared_ptr<tf::Transformer>& tf,
                                                                      ros::NodeHandle nh)
 {
   SharedStorage& s = getSharedStorage();
   boost::mutex::scoped_lock slock(s.lock_);
-  if (s.state_monitors_.find(kmodel->getName()) != s.state_monitors_.end())
-    return s.state_monitors_[kmodel->getName()];
+  if (s.state_monitors_.find(robot_description) != s.state_monitors_.end())
+    return s.state_monitors_[robot_description];
```

Comment:
this is a big change in behavior.. getName() usually returns e.g. ``pr2`` whereas robot_description ``robot_description``

Reply:
I made such a change because of passing the included test code.
This test code makes it possible to control two robots via moveit.
In the test code, it sets two robot models to rosparam.
The first one is "/robot_description", the second one is "/test/robot_description".

In the original code, `kmodel->getName()` of the both robots becomes "fanuc", and it returns the same state to the two robots.
In the changed code, I use `robot_description` so that keys are unique.

Reply:
I am also aware that it is a big change, but I definitely want to control multiple robots with moveit.
I would like to try it if you have an alternative idea.